servicex/databinder/databinder_configuration.py

```diff
@@ -0,0 +1,257 @@
+                        if 'DEF_' in value:
+                            def_in_value = True
+                            for repre, new_str in definition.items():
+                                if repre == value:
+                                    samples[n][field] \
+                                        = samples[n][field] \
+                                        .replace(repre, new_str)
+                                    ndef = ndef + 1
+                                    def_in_value = False
+                            if def_in_value:
+                                raise NotImplementedError(
+                                    f"{value} is NOT defined in the Definition block"
+                                )
+        return ndef
+    else:
+        return ndef
+
+
+def _support_old_option_names(config: Dict[str, Any]) -> Dict:
+    """ """
+    if 'ServiceXName' in config['General'].keys():
+        config['General']['ServiceX'] = config['General'].pop('ServiceXName')
+
+    if 'Transformer' in config['General'].keys():
+        config['General']['Codegen'] = config['General'].pop('Transformer')
+    for (idx, sample) in zip(range(len(config['Sample'])), config['Sample']):
+        if 'Transformer' in sample.keys():
+            config['Sample'][idx]['Codegen'] = config['Sample'][idx].pop('Transformer')
+
+    if 'IgnoreServiceXCache' in config['General'].keys():
+        config['General']['IgnoreLocalCache'] = config['General'].pop('IgnoreServiceXCache')
+    for (idx, sample) in zip(range(len(config['Sample'])), config['Sample']):
+        if 'IgnoreServiceXCache' in sample.keys():
+            config['Sample'][idx]['IgnoreLocalCache'] = config['Sample'][idx].pop(
+                'IgnoreServiceXCache')
+    return config
+
+
+def _decorate_backend_per_sample(config: Dict[str, Any]) -> Dict:
+    """ from General block """
+    pair = ("default_transformer", "default_codegen")
+    if 'Codegen' in config['General'].keys():
+        if config['General']['Codegen'] == "atlasr21":
+            pair = ("xaod", "atlasr21")
+        elif config['General']['Codegen'] == "uproot":
+            pair = ("uproot", "uproot")
+        elif config['General']['Codegen'] == "python":
+            pair = ("uproot", "python")
+
+    """ from Sample block """
+    for (idx, sample) in zip(range(len(config['Sample'])), config['Sample']):
+        if 'Codegen' in sample.keys():
+            if sample['Codegen'] == "atlasr21":
+                config['Sample'][idx]['Transformer'] = "xaod"
+            elif sample['Codegen'] == "uproot":
+                config['Sample'][idx]['Transformer'] = "uproot"
+            elif sample['Codegen'] == "python":
+                config['Sample'][idx]['Transformer'] = "uproot"
+        else:
+            config['Sample'][idx]['Transformer'] = pair[0]
+            config['Sample'][idx]['Codegen'] = pair[1]
+
+    return config
+
+
+def _validate_config(config: Dict[str, Any]):
```

Comment:
It would be so much simpler to replace this generic dict with a [pydantic](https://docs.pydantic.dev/latest/) model (see models.py)